src/gui/pedalboard/cables/CableView.cpp

```diff
@@ -15,31 +15,56 @@ CableView::CableView (BoardComponent* comp) : board (comp)
 
 CableView::~CableView() = default;
 
-void CableView::paint (Graphics& g)
+bool CableView::MouseOverClickablePort()
 {
-    using namespace CableDrawingHelpers;
-
+    const auto mousePos = Desktop::getMousePosition() - getScreenPosition();
+    nearestPort = portLocationHelper->getNearestPort (mousePos, board);
     if (nearestPort.editor != nullptr)
     {
         const bool isDraggingNearInputPort = nearestPort.isInput && isDraggingCable;
         const bool isNearConnectedInput = nearestPort.isInput && ! portLocationHelper->isInputPortConnected (nearestPort);
         if (! (isDraggingNearInputPort || isNearConnectedInput))
         {
-            auto startPortLocation = CableViewPortLocationHelper::getPortLocation (nearestPort);
-            drawCablePortGlow (g, startPortLocation, scaleFactor);
+            return true;
         }
     }
 
+    return false;
+}
+
+bool CableView::MouseDraggingOverOutputPort()
+{
     if (cableMouse != nullptr)
     {
         auto mousePos = cableMouse->getPosition();
         const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cables.getLast()->connectionInfo.startProc);
         if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
         {
-            auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
-            drawCablePortGlow (g, endPortLocation, scaleFactor);
+            return true;
         }
     }
+
+    return false;
+}
+
+void CableView::paint (Graphics& g)
+{
+    using namespace CableDrawingHelpers;
+
+    if (MouseOverClickablePort())
+    {
+        auto startPortLocation = CableViewPortLocationHelper::getPortLocation (nearestPort);
+        drawCablePortGlow (g, startPortLocation, scaleFactor);
+        portGlowOn = true;
+    }
+    else if (MouseDraggingOverOutputPort())
+    {
+        auto mousePos = cableMouse->getPosition();
+        const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cables.getLast()->connectionInfo.startProc);
```

Comment:
I wish we could save `nearestInputPort` from `mouseDraggingOverOutputPort()`, since it takes a good bit of work to find it again.